huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_resource_share_associations_test.go

```diff
@@ -0,0 +1,194 @@
+					resource.TestCheckResourceAttrSet(dataSourcePrincipal, "associations.0.status"),
+					resource.TestCheckResourceAttrSet(dataSourcePrincipal, "associations.0.updated_at"),
+
+					dcTestResource.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.#"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.associated_entity"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.association_type"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.created_at"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.external"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.resource_share_id"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.resource_share_name"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.status"),
+					resource.TestCheckResourceAttrSet(dataSourceResource, "associations.0.updated_at"),
+
+					dcByPrincipal.CheckResourceExists(),
+					resource.TestCheckOutput("is_principal_filter_useful", "true"),
+
+					dcByResourceShareIds.CheckResourceExists(),
+					resource.TestCheckOutput("is_resource_share_ids_filter_useful", "true"),
+
+					dcByResourceUrn.CheckResourceExists(),
+					resource.TestCheckOutput("is_resource_urn_filter_useful", "true"),
+
+					dcByStatus.CheckResourceExists(),
+					resource.TestCheckOutput("is_status_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceAssociations_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+# Filter by principal association type. Field association_type is required.
+data "huaweicloud_ram_resource_share_associations" "test_principal" {
+  depends_on = [huaweicloud_ram_resource_share.test]
+
+  association_type = "principal"
+}
+
+# Filter by resource association type. Field association_type is required.
+data "huaweicloud_ram_resource_share_associations" "test_resource" {
+  depends_on = [huaweicloud_ram_resource_share.test]
+
+  association_type = "resource"
+}
+
+# Filter by principal
+locals {
+  principal = data.huaweicloud_ram_resource_share_associations.test_principal.associations[0].associated_entity
+}
+
+data "huaweicloud_ram_resource_share_associations" "filter_by_principal" {
+  depends_on = [huaweicloud_ram_resource_share.test]
+
+  association_type = "principal"
+  principal        = local.principal
+}
+
+locals {
+  principal_filter_result = [
+    for v in data.huaweicloud_ram_resource_share_associations.filter_by_principal.associations[*].associated_entity :
+v == local.principal
+  ]
+}
+
+output "is_principal_filter_useful" {
+  value = length(local.principal_filter_result) > 0 && alltrue(local.principal_filter_result)
+}
+
+# Filter by resource_share_ids
+locals {
+  resource_share_ids = split(",",
+data.huaweicloud_ram_resource_share_associations.test_principal.associations[0].resource_share_id)
```

Comment:
Ditto. Please also check for formatting errors elsewhere

Reply:
ok